spec_summary/filter_map_test.go

```diff
@@ -12,14 +14,15 @@ func TestFiltersToMethods(t *testing.T) {
 
 	m1 := akid.GenerateAPIMethodID().GetUUID().String()
 	m2 := akid.GenerateAPIMethodID().GetUUID().String()
+	ms := sets.NewSet(m1, m2)
 
 	fm.InsertNondirectionalFilter(HostFilter, "example.com", m1)
 	fm.InsertNondirectionalFilter(HostFilter, "example.com", m2)
-	fm.InsertDirectionalFilter(RequestDirection, AuthFilter, "None", m1)
-	fm.InsertDirectionalFilter(RequestDirection, AuthFilter, "None", m2)
+	fm.InsertDirectionalFilter(RequestDirection, AuthFilter, "None", m1, ms)
+	fm.InsertDirectionalFilter(RequestDirection, AuthFilter, "None", m2, ms)
 
-	fm.filterMap.Insert(HttpMethodFilter, "GET", m1)
-	fm.filterMap.Insert(HttpMethodFilter, "POST", m2)
+	fm.filterMap.Insert(HttpMethodFilter, "GET", optionals.Some(m1))
+	fm.filterMap.Insert(HttpMethodFilter, "POST", optionals.Some(m2))
 
 	// No filters.
 	directedSummary, numMethods := fm.SummarizeWithFilters(nil)
```

Comment:
I am kind of surprised that the "with filters" case below did not result in any changes, but it was checking that the counts were equal to zero, which is what we'd expect in any case.

Is there an assert we can add to distinguish the new and old behavior?

Reply:
Hm, maybe it's because we didn't check the MethodID behavior at all?  In which case some test would be appropriate for that.

Reply:
New tests in 3396096.